diff --git a/node_modules/@react-native-camera-roll/camera-roll/ios/RNCCameraRoll.mm b/node_modules/@react-native-camera-roll/camera-roll/ios/RNCCameraRoll.mm
index d63f1a7..fea9496 100644
--- a/node_modules/@react-native-camera-roll/camera-roll/ios/RNCCameraRoll.mm
+++ b/node_modules/@react-native-camera-roll/camera-roll/ios/RNCCameraRoll.mm
@@ -766,12 +766,26 @@ RCT_EXPORT_METHOD(getPhotoByInternalID:(NSString *)internalId
 
             filePath = [imageURL.absoluteString stringByReplacingOccurrencesOfString:@"pathfile:" withString:@"file:"];
 
+            
+              // Convert the filePath (NSString) back to NSURL
+              NSURL *updatedURL = [NSURL URLWithString:filePath];
+
+              // Use NSURLComponents to remove the fragment
+              NSURLComponents *components = [NSURLComponents componentsWithURL:updatedURL resolvingAgainstBaseURL:NO];
+              components.fragment = nil; // Remove the fragment part of the URL
+
+              // Get the corrected file path as a string
+              NSString *correctPath = components.URL.absoluteString;
+
+              // Log or use the correctPath
+              NSLog(@"Corrected Path: %@", correctPath);
+              
             resolve(@{
                       @"node": @{
                           @"type": assetMediaTypeLabel,
                           @"subTypes":assetMediaSubtypesLabel,
                           @"image": @{
-                              @"filepath": filePath,
+                              @"filepath": correctPath,
                               @"filename": originalFilename,
                               @"height": @([asset pixelHeight]),
                               @"width": @([asset pixelWidth]),
