diff --git a/node_modules/react-native-tab-view/src/PlatformPressable.tsx b/node_modules/react-native-tab-view/src/PlatformPressable.tsx
index 1e1b18d..0e46d25 100644
--- a/node_modules/react-native-tab-view/src/PlatformPressable.tsx
+++ b/node_modules/react-native-tab-view/src/PlatformPressable.tsx
@@ -20,18 +20,13 @@ const ANDROID_SUPPORTS_RIPPLE =
  */
 export function PlatformPressable({
   android_ripple,
-  pressColor = 'rgba(0, 0, 0, .32)',
+  pressColor = 'rgba(0, 0, 0, 0)',
   pressOpacity,
   style,
   ...rest
 }: Props) {
   return (
     <Pressable
-      android_ripple={
-        ANDROID_SUPPORTS_RIPPLE
-          ? { color: pressColor, ...android_ripple }
-          : undefined
-      }
       style={({ pressed }) => [
         { opacity: pressed && !ANDROID_SUPPORTS_RIPPLE ? pressOpacity : 1 },
         typeof style === 'function' ? style({ pressed }) : style,
