diff --git a/node_modules/react-native-camera/android/build/tmp/compileGeneralDebugJavaWithJavac/previous-compilation-data.bin b/node_modules/react-native-camera/android/build/tmp/compileGeneralDebugJavaWithJavac/previous-compilation-data.bin
new file mode 100644
index 0000000..d866b14
Binary files /dev/null and b/node_modules/react-native-camera/android/build/tmp/compileGeneralDebugJavaWithJavac/previous-compilation-data.bin differ
diff --git a/node_modules/react-native-camera/src/RNCamera.js b/node_modules/react-native-camera/src/RNCamera.js
index b7a271a..4c9badc 100644
--- a/node_modules/react-native-camera/src/RNCamera.js
+++ b/node_modules/react-native-camera/src/RNCamera.js
@@ -1,17 +1,16 @@
 // @flow
-import React from 'react';
 import PropTypes from 'prop-types';
+import React from 'react';
 import {
-  findNodeHandle,
-  Platform,
-  NativeModules,
-  ViewPropTypes,
-  requireNativeComponent,
-  View,
   ActivityIndicator,
-  Text,
-  StyleSheet,
+  NativeModules,
   PermissionsAndroid,
+  Platform,
+  StyleSheet,
+  Text,
+  View,
+  findNodeHandle,
+  requireNativeComponent,
 } from 'react-native';
 
 import type { FaceFeature } from './FaceDetector';
@@ -394,7 +393,6 @@ export default class Camera extends React.Component<PropsType, StateType> {
   };
 
   static propTypes = {
-    ...ViewPropTypes,
     zoom: PropTypes.number,
     useNativeZoom: PropTypes.bool,
     maxZoom: PropTypes.number,
